### DISCUSSION TITLE
Able to config random suffix format (for example 6 random string characters)

### DISCUSSION THREAD

**Discussion by nguyenkims** [simple-login/app#426] [Feature Requests]
Moved from https://trello.com/c/LHo5vunB/161-able-to-config-random-suffix-format-for-example-6-random-string-characters

**Comment by developStorm** [comment]
I would probably be able to implement this. Anything I need to know before get started?

**Reply by nguyenkims** [reply]
Hey thanks a lot of offering to implement this! The method that returns a list of available suffix is `get_available_suffixes` that I think we can change to add the 6 random character option. https://github.com/simple-login/app/blob/ceacf8e3a71262abdfa2e47a25adc7c868d4789c/app/dashboard/views/custom_alias.py#L44

I think it's best if we can have an option so user can opt out of the 6-character option (and therefore stays with the current random word approach) as some people prefer the word option :).

**Reply by developStorm** [reply]
Cool. What format of string would you think work best? Should it be all letters, or letter + numbers?

**Reply by developStorm** [reply]
I figured this feature might conflict with `verify_prefix_suffix`: 
https://github.com/simple-login/app/blob/ceacf8e3a71262abdfa2e47a25adc7c868d4789c/app/dashboard/views/custom_alias.py#L285

This function is checking to make sure suffix is in the word list, while the random string couldn't be in it apparently. I'm a bit confused as to why this check is necessary, it seems that the current suffix has been signed and I don't see the point of verifying if it's in list again ...... The easiest way might be to just remove this part of the check, what do you think?

**Reply by nguyenkims** [reply]
Hey the reason this check was still there because the below code is still used by our safari extension and the extension doesn't send the signed suffix. We're going to update the code in Safari extension, leave a few days (1 week?) so all users have their extension upgraded and then we can delete this `word_exist` check :)

https://github.com/simple-login/app/blob/d5de99afe964fd7838dd53d650616ac59a786d08/app/api/views/new_custom_alias.py#L33

**Comment by developStorm** [comment]
Huh the PR #473 did not automatically associate to here

**Reply by nguyenkims** [reply]
Yeah somehow "GitHub discussions" doesn't work the same way as "Issues" ... Hope they will converge in the future as discussions is better than Issues on most areas :)
